crypto/hash/src/main/java/org/apache/shiro/crypto/hash/ConfigurableHashService.java

```diff
@@ -18,44 +18,20 @@
  */
 package org.apache.shiro.crypto.hash;
 
-import org.apache.shiro.crypto.RandomNumberGenerator;
-import org.apache.shiro.lang.util.ByteSource;
-
 /**
  * A {@code HashService} that allows configuration of its strategy via JavaBeans-compatible setter methods.
  *
  * @since 1.2
  */
 public interface ConfigurableHashService extends HashService {
 
-    /**
-     * Sets the 'private' (internal) salt to be paired with a 'public' (random or supplied) salt during hash computation.
-     *
-     * @param privateSalt the 'private' internal salt to be paired with a 'public' (random or supplied) salt during
-     *                    hash computation.
-     */
-    void setPrivateSalt(ByteSource privateSalt);
-
-    /**
-     * Sets the number of hash iterations that will be performed during hash computation.
-     *
-     * @param iterations the number of hash iterations that will be performed during hash computation.
-     */
-    void setHashIterations(int iterations);
-
     /**
      * Sets the name of the {@link java.security.MessageDigest MessageDigest} algorithm that will be used to compute
      * hashes.
      *
      * @param name the name of the {@link java.security.MessageDigest MessageDigest} algorithm that will be used to
      *             compute hashes.
      */
-    void setHashAlgorithmName(String name);
+    void setDefaultAlgorithmName(String name);
```

Comment:
nit: might need to tweak the javadoc to go with this, I'm assuming this is the alg used if the hash cannot be determined?

Reply:
Yes, the javadocs are the unchecked checkbox ;) Next commit is on the way.

The default is being used when a `HashRequest` does not carry an algorithm name. Its old behaviour/semantics was at least similar, but I was not looking into making it a 100% compatible here.